jobs/src/main/scala/dcos/metronome/utils/glue/MarathonImplicits.scala

```diff
@@ -72,7 +72,8 @@ object MarathonImplicits {
 
   implicit class JobSpecToContainer(val jobSpec: JobSpec) extends AnyVal {
     def toContainer: Option[Container] = {
-      jobSpec.run.docker match {
+      require(!(jobSpec.run.docker.nonEmpty && jobSpec.run.ucr.nonEmpty), "docker and ucr can't be present both")
```

Comment:
I like the idea of failing loud and proud here, though I don't like the fact that we actually have to _do_ this kind of check. Ideally I would like to see a `DockerJobSpec` and a `UCRJobSpec`, bot deriving from the same `JobSpec`.

Nonetheless, this is working right now and we could add this on tech debt on a next revision, so I am not going to block the PR.